src/System.IO.Ports/src/System/IO/Ports/SerialPort.Uap.cs

```diff
@@ -10,8 +10,39 @@ public partial class SerialPort : Component
     {
         public static string[] GetPortNames()
         {
-            // See https://github.com/dotnet/corefx/issues/20588 for more information.
-            throw new PlatformNotSupportedException(System.SR.PlatformNotSupported_SerialPort_GetPortNames);
+            Span<uint> portNumbers = stackalloc uint[16];
+            uint portNumbersFound;
+            int error;
+
+            try
+            {
+                error = Interop.mincore.GetCommPorts(portNumbers, out portNumbersFound);
+            }
+            catch (Exception e) when (e is EntryPointNotFoundException || e is DllNotFoundException)
+            {
+                throw new PlatformNotSupportedException(System.SR.PlatformNotSupported_SerialPort_GetPortNames);
+            }
+
+            if (error == Interop.Errors.ERROR_MORE_DATA)
+            {
+                portNumbers = portNumbersFound <= 64 ?
```

Comment:
This is premature optimization - unnecessary code bloat. Just always allocate it on the heap.